tests/foreman/ui/test_discoveryrule.py

```diff
@@ -21,26 +21,24 @@
 from fauxfactory import gen_integer
 from fauxfactory import gen_ipaddr
 from fauxfactory import gen_string
-from nailgun import entities
 
 
 @pytest.fixture(scope='module')
-def manager_loc():
-    return entities.Location().create()
-
-
-@pytest.fixture(scope='module')
-def module_org():
-    return entities.Organization().create()
+def manager_loc(module_target_sat):
+    return module_target_sat.api.Location().create()
 
 
 @pytest.fixture
-def module_discovery_env(module_org, module_location):
-    discovery_loc = entities.Setting().search(query={'search': 'name="discovery_location"'})[0]
+def module_discovery_env(module_org, module_location, module_target_sat):
+    discovery_loc = module_target_sat.api.Setting().search(
+        query={'search': 'name="discovery_location"'}
+    )[0]
     default_discovery_loc = discovery_loc.value
     discovery_loc.value = module_location.name
     discovery_loc.update(['value'])
```

Comment:
How about using `update_setting` helper here instead? it returns the default values which can later be used for teardown. Same for Org and Loc
```suggestion
    discovery_loc = module_target_sat.update_setting('discovery_location', module_location.name)
    yield
    module_target_sat.update_setting('discovery_location', discovery_loc)
```